gui/app/main.cpp

```diff
@@ -16,12 +18,37 @@
 #include <iostream>
 #include <memory>
 
+#include "batteryinfo.h"
+#include "simple_clock.h"
+
+QObject *gui_state_instance(QQmlEngine *engine, QJSEngine *scriptEngine) {
+  static GuiStateContainer state_container(
```

Comment:
Please add a comment explaining why we maintain this in a static variable and why do we indicate CppOwnership below (a static object can't really be owned, right?)

Reply:
Added the following comment:
```
  // Since we are returning just a pointer, QQmlEngine does not know the object
  // is static and will try to take ownership of this object, unless we manually
  // tell the engine that the ownership will belong to Cpp side.
```